pkg/kubeclient/signup.go

```diff
@@ -77,27 +83,42 @@ func (c *userSignupClient) Update(obj *crtapi.UserSignup) (*crtapi.UserSignup, e
 	return result, nil
 }
 
-func (c *userSignupClient) ListByPhone(phone string) (*crtapi.UserSignupList, error) {
-	return c.listByHashedLabel(crtapi.BannedUserPhoneNumberHashLabelKey, phone)
-}
+func (c *userSignupClient) ListByPhone(value string) (*crtapi.UserSignupList, error) {
+	if e164Matcher.Match([]byte(value)) {
+		return c.listByHashedLabelValue(crtapi.BannedUserPhoneNumberHashLabelKey, value)
+	}
+
+	if md5Matcher.Match([]byte(value)) {
+		return c.listByLabel(crtapi.BannedUserPhoneNumberHashLabelKey, value)
+	}
 
-// ListByHashedLabel returns a UserSignupList containing any UserSignup resources that have a label matching the specified label
-func (c *userSignupClient) listByHashedLabel(labelKey, labelValue string) (*crtapi.UserSignupList, error) {
+	// Default to searching for a hash of the specified value
+	return c.listByHashedLabelValue(crtapi.BannedUserPhoneNumberHashLabelKey, value)
+}
 
-	// Calculate the md5 hash for the phoneNumber
+// listByHashedLabelValue returns a UserSignupList containing any UserSignup resources that have a label matching the
+// md5 hash of the specified value
+func (c *userSignupClient) listByHashedLabelValue(labelKey, value string) (*crtapi.UserSignupList, error) {
```

Comment:
But `value` can be already hashed, right?

Reply:
This function calculates the hash of the provided value and searches for that.  I'm not a fan of the function name but couldn't think of a better one in the moment.

Reply:
OK. I re-read the code and comments and I think it's clear enough. Let's keep it as is.